BMM.Core/Implementations/Networking/ExperimentIdHeaderProvider.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using BMM.Api.Abstraction;
+using BMM.Core.Implementations.FirebaseRemoteConfig;
+
+namespace BMM.Core.Implementations.Networking
+{
+    public class ExperimentIdHeaderProvider : IHeaderProvider
+    {
+        private readonly IFirebaseRemoteConfig _config;
+
+        public ExperimentIdHeaderProvider(IFirebaseRemoteConfig config)
+        {
+            _config = config;
+        }
+
+        public async Task<KeyValuePair<string, string>?> GetHeader()
+        {
+            if (!string.IsNullOrEmpty(_config.ExperimentId))
+            {
+                return new KeyValuePair<string, string>("ExperimentId", _config.ExperimentId);
```

Comment:
that's a bit tricky. In general, I agree that using nameof is better. But in this case I want to make sure, that the string is never changed, since the API expects it to be ExperimentId. Right now, you can refactor the name without breaking other systems.

Reply:
extracted all header names